### ISSUE TITLE
Consider adding short keys for JSON encoding

### ISSUE THREAD

**Issue by tigrannajaryan** [open-telemetry/opentelemetry-proto#412]
`json_name` option can be used to shorten keys for JSON encoding. This can result in significant size savings for uncompressed JSON, which appears to be important for e.g. https://github.com/open-telemetry/oteps/pull/208#discussion_r904191239

**Comment by tigrannajaryan** [comment]
Here is size comparison using one or two letter JSON keys:
```
===== Encoded sizes
Encoding                        Uncomp/json[Improved]   zlib/json[Improved]
OTLP/Logs                       55005 by   [1.000]      4627 by [1.000]
ShortKeys/Logs                  41768 by   [1.317]      4386 by [1.055]

Encoding                        Uncomp/json[Improved]   zlib/json[Improved]
OTLP/Trace/Attribs              41738 by   [1.000]      3513 by [1.000]
ShortKeys/Trace/Attribs         31998 by   [1.304]      3356 by [1.047]

Encoding                        Uncomp/json[Improved]   zlib/json[Improved]
OTLP/Trace/Events               49336 by   [1.000]      2572 by [1.000]
ShortKeys/Trace/Events          34396 by   [1.434]      2435 by [1.056]

Encoding                        Uncomp/json[Improved]   zlib/json[Improved]
OTLP/Metric/Histogram           37993 by   [1.000]      1809 by [1.000]
ShortKeys/Metric/Histogram      25171 by   [1.509]      1641 by [1.102]

Encoding                        Uncomp/json[Improved]   zlib/json[Improved]
OTLP/Metric/MixOne             149353 by   [1.000]      4185 by [1.000]
ShortKeys/Metric/MixOne        104231 by   [1.433]      3894 by [1.075]

Encoding                        Uncomp/json[Improved]   zlib/json[Improved]
OTLP/Metric/MixSeries          549540 by   [1.000]     13819 by [1.000]
ShortKeys/Metric/MixSeries     395682 by   [1.389]     13103 by [1.055]
```

The uncompressed JSON improvement with short keys is around 1.3-1.5 times compared to the regular keys, which use full field names.

**Comment by Oberon00** [comment]
We could even use the numeric values (stringized) as JSON keys (decimal or hex or base64). Then we would also get more stable / renaming-resistant keys.

**Comment by tigrannajaryan** [comment]
To summarize the findings in [the PR](https://github.com/open-telemetry/opentelemetry-proto/pull/413):
- There is an option to do gzip compression using [CompressionStreams API](https://wicg.github.io/compression/) available in Chromium only.
- There is an option to use binary Protobuf, and https://github.com/protobufjs/protobuf.js/ is 6.5KB and claims performance faster than native JSON. Protobuf is going to be more compact than JSON with short keys (about 1.6-2.2 in my benchmarks).

We also discussed this in the Spec meeting yesterday and no-one in the call believed that we need the JSON with short keys given that other options are available. JSON with short keys loses its readability, while not gaining as much in the performance as the other options we have.

Given the above, I am marking this issue as "won't do". If there are new arguments the issue can be re-opened (but it must happen before we declare JSON format Stable).
